### ISSUE TITLE
[BUG] `anchor-name` doesn't respect CSS cascade

### ISSUE THREAD

**Issue by mmalerba** [oddbird/css-anchor-positioning#218]
**Describe the bug**
The `anchor-name` property doesn't respect the CSS cascade. For example:

```html
.anchor {
  anchor-name: --name;
}
.anchor {
  anchor-name: --other-name;
}
.positioned {
  position: absolute;
  top: anchor(--name, bottom); // this should not anchor to `.anchor`, but currently does.
}
```

**To Reproduce**
I created a test demonstrating the issue here: https://github.com/oddbird/css-anchor-positioning/compare/main...mmalerba:css-anchor-positioning:cascade

**Expected behavior**
The `anchor` function should only match the `anchor-name` that is actually applied by the cascade

**Additional context**
This can be solved by shifting `anchor-name` into a custom property, but the custom property inherits by default, so we would either have to disable that with `@property` (potentially unacceptable due to browser support?) or just be careful to verify we're not reading inherited values. This is how I handled a similar issue for `anchor-scope`: [`32f04d8` (#217)](https://github.com/oddbird/css-anchor-positioning/pull/217/commits/32f04d88e745829ba0b2fcd42f98612ad5c9fdea)

**Comment by jgerigmeyer** [comment]
Good catch! I like the approach you used in #217, so that would be my leaning.

**Comment by jgerigmeyer** [comment]
I think this solution might also fix #212?
